source/auth/tests/mongodb-oidc.rst

```diff
@@ -2,458 +2,435 @@
+- Create a ``MongoClient`` configured with an OIDC callback that validates its
+  inputs and returns a valid access token.
 - Perform a ``find`` operation that succeeds.
+- Assert that the OIDC callback was called with the appropriate inputs,
+  including the timeout parameter if possible.
 - Close the client.
 
-Multiple Principal User 2
-~~~~~~~~~~~~~~~~~~~~~~~~~
-- Set the ``AWS_WEB_IDENTITY_TOKEN_FILE`` environment variable
-  to the location of valid ``test_user2`` credentials.
-- Create a client with a url of the form ``mongodb://localhost:27018/?authMechanism=MONGODB-OIDC&authMechanismProperties=PROVIDER_NAME:aws&directConnection=true&readPreference=secondaryPreferred``.
-- Perform a ``find`` operation that succeeds.
-- Close the client.
+**2.2 OIDC Callback Returns Null**
 
-Allowed Hosts Ignored
-~~~~~~~~~~~~~~~~~~~~~
-- Create a client with a url of the form ``mongodb://localhost/?authMechanism=MONGODB-OIDC&authMechanismProperties=PROVIDER_NAME:aws``, and an
-  ``ALLOWED_HOSTS`` that is an empty list.
-- Assert that a ``find`` operation succeeds.
+- Create a ``MongoClient`` configured with an OIDC callback that returns
+  ``null``.
+- Perform a ``find`` operation that fails.
 - Close the client.
 
-Callback Validation
-===================
-
-Valid Callbacks
-~~~~~~~~~~~~~~~
-- Clear the cache.
-- Create request and refresh callback that validate their inputs and return
-  a valid token.  The request callback must return a token that expires in
-  one minute.
-- Create a client that uses the above callbacks.
-- Perform a ``find`` operation that succeeds.  Verify that the request
-  callback was called with the appropriate inputs, including the timeout
-  parameter if possible.  Ensure that there are no unexpected fields.
-- Close the client.
-- Create a new client with the same configuration.
-- Perform a ``find`` operation that succeeds.  Verify that the refresh
-  callback was called with the appropriate inputs, including the timeout
-  parameter if possible.
-- Close the client.
+**2.3 OIDC Callback Returns Missing Data**
 
-Request Callback Returns Null
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-- Clear the cache.
-- Create a client with a request callback that returns ``null``.
+- Create a ``MongoClient`` configured with an OIDC callback that returns data
+  not conforming to the ``OIDCCredential`` with missing fields.
 - Perform a ``find`` operation that fails.
 - Close the client.
 
-Refresh Callback Returns Null
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-- Clear the cache.
-- Create request callback that returns a valid token that will expire in a
-  minute, and a refresh callback that returns ``null``.
+**2.4 Invalid Client Configuration with Callback**
+
+- Create a ``MongoClient`` configured with an OIDC callback and auth mechanism
+  property ``PROVIDER_NAME:aws``.
+- Assert it returns a client configuration error.
```

Comment:
This prose test is already covered by the unified connection string tests above. Isn't this an unnecessary duplicate?

Reply:
The updated Auth connection string tests don't support passing an OIDC callback, so I believe this prose test covers a case that the Auth connection string tests can't. The previous Auth connection string OIDC tests did include a "callback" param, but it was not documented in the test format README, was only implemented in 1 or 2 drivers, and was confusing with the updated OIDC callback API, so I decided to replace it with a prose test instead.